lua/luasnip/extras/otf.lua

```diff
@@ -0,0 +1,86 @@
+local ls = require("luasnip")
+local cp = require("luasnip.util.functions").copy
+local p = require("luasnip.extras._parser_combinator")
+
+local T = { EOL = "EOL", TXT = "TXT", INP = "INP" }
+
+local chunk = p.any(
+	p.map(p.literal("$$"), function()
+		return { T.TXT, "$" }
+	end),
+	p.map(p.literal("\n"), function()
+		return { T.EOL }
+	end),
+	p.map(p.seq(p.literal("$"), p.pattern("%w+")), function(c)
+		return { T.INP, c[1] }
+	end),
+	p.map(p.pattern("[^\n$]*"), function(c)
+		return { T.TXT, c }
+	end)
+)
+
+local snippet_chunks = p.star(chunk)
+
+local function txt_to_snip(txt)
+	local t = ls.t
+	local sn = ls.s
+	local i = ls.i
+	local f = ls.f
+
+	-- The parser does not handle empty strings
+	if txt == "" then
+		return sn(t({ "" }))
+	end
+
+	local _, chunks, _ = snippet_chunks(txt, 1)
+
+	local current_text_arg = { "" }
+	local nodes = {}
+	local know_inputs = {}
+	local last_input_pos = 0
+
+	for _, part in ipairs(chunks) do
+		if part[1] == T.TXT then
+			current_text_arg[#current_text_arg] = current_text_arg[#current_text_arg]
+				.. part[2]
+		elseif #current_text_arg > 0 then
```

Comment:
IIUC this is for committing merged TXT to the nodes, right? If so, it would generate some empty textNodes because the empty `current_text_arg` is `{""}`, which matches the `if`.
Checking for length > 1 or [1] ~= "" should work (?)